conan/tools/microsoft/__init__.py

```diff
@@ -2,3 +2,17 @@
 from .toolchain import MSBuildToolchain
 from .msbuild import MSBuild
 from .msbuilddeps import MSBuildDeps
+
+
+def msvc_runtime_flag(conanfile):
+    settings = conanfile.settings
+    compiler = settings.get_safe("compiler")
+    runtime = settings.get_safe("compiler.runtime")
```

Comment:
```suggestion
    runtime = settings.get_safe("compiler.base.runtime") or settings.get_safe("compiler.runtime")
```

Reply:
If this is for the intel compiler, I prefer to wait until the next iteration. For a start, the ``settings.yml`` needs to be changed too, this will not work without changing the settings. And we need to understand how the binary model for intel will work, now that we are defining a fallback from msvc => Visual Studio, because intel compiler was relying on that too, and it might have some collisions.

Reply:
again, that might not be not just Intel compiler, but any custom compiler with custom settings using `msvc` as a `compiler.base`.